src/lib/drivers/gyroscope/PX4Gyroscope.cpp

```diff
@@ -236,38 +260,21 @@ void PX4Gyroscope::updateFIFO(const FIFOSample &sample)
 
 
 	// clipping
-	const int16_t clip_limit = (_range / _scale) * 0.95f;
-
-	// x clipping
-	for (int n = 0; n < sample.samples; n++) {
-		if (abs(sample.x[n]) > clip_limit) {
-			_clipping[0]++;
-			_integrator_clipping++;
-		}
-	}
-
-	// y clipping
-	for (int n = 0; n < sample.samples; n++) {
-		if (abs(sample.y[n]) > clip_limit) {
-			_clipping[1]++;
-			_integrator_clipping++;
-		}
-	}
+	unsigned clip_count_x = clipping(sample.x, _clip_limit, N);
+	unsigned clip_count_y = clipping(sample.y, _clip_limit, N);
+	unsigned clip_count_z = clipping(sample.y, _clip_limit, N);
```

Comment:
Typo (sample.y in row where you should have z).

Reply:
Nice catch, fixed.